src/abstractdataframe/abstractdataframe.jl

```diff
@@ -943,17 +943,25 @@ without(df::AbstractDataFrame, c::Any) = without(df, index(df)[c])
 
 # catch-all to cover cases where indexing returns a DataFrame and copy doesn't
 
-Base.hcat(df::AbstractDataFrame, x; makeunique::Bool=false) =
-    hcat!(copy(df), x, makeunique=makeunique)
-Base.hcat(x, df::AbstractDataFrame; makeunique::Bool=false) =
-    hcat!(x, df, makeunique=makeunique)
-Base.hcat(df1::AbstractDataFrame, df2::AbstractDataFrame; makeunique::Bool=false) =
-    hcat!(copy(df1), df2, makeunique=makeunique)
-Base.hcat(df::AbstractDataFrame, x, y...; makeunique::Bool=false) =
-    hcat!(hcat(df, x, makeunique=makeunique), y..., makeunique=makeunique)
+Base.hcat(df::AbstractDataFrame, copycolumns::Bool=true) =
+    copycolumns ? copy(df) : DataFrame(eachcol(df, false), names(df))
```

Comment:
@nalimilan: additionally here and below it is not clear what we should return if `copycols=false`.

Reply:
What's wrong with what you do now?

Reply:
I force it to be `DataFrame` and in general I can imagine some other `AbstractDataFrame` subtype that should produce this subtype after `hcat` (of course the user can override `hcat` for that type and `DataFrame` can be the default as it is now, but this solution is not very clean)

Reply:
OK. Not the end of the world I guess. Something with `Tables.materializer` would probably be slower?